From 1dc5bf780f19bb9dc3a0ac1ae99a4182f4a6f289 Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@digia.com>
Date: Tue, 17 Dec 2013 12:26:58 +0100
Subject: [PATCH 18/20] Fix QtWebKit build on ARM softfp

The low level interpreter depends on hardfp, so we need to disable it
when builing on ARM targets with softfp.

Task-number: QTBUG-35681
Change-Id: I8aaef546b92182a87cc7116ac452d8725de6a7db
Reviewed-by: Jocelyn Turcotte <jocelyn.turcotte@digia.com>

Conflicts:
	Source/WTF/wtf/Platform.h
---
 Source/WTF/wtf/Platform.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/Source/WTF/wtf/Platform.h b/Source/WTF/wtf/Platform.h
index 5782ce4..af2b8a3 100644
--- a/Source/WTF/wtf/Platform.h
+++ b/Source/WTF/wtf/Platform.h
@@ -911,6 +911,16 @@
 #define ENABLE_DISASSEMBLER 1
 #endif
 
+/* Disable the LLINT on versions of GCC prior to 4.3. Mainly due to buggy assembler on OSX 10.6, the only supported platform using that old a version. */
+#if !defined(ENABLE_LLINT) && COMPILER(GCC) && !GCC_VERSION_AT_LEAST(4, 3, 0)
+#define ENABLE_LLINT 0
+#endif
+
+/* LLINT on ARM depends on an FPU */
+#if !defined(ENABLE_LLINT) && CPU(ARM) && !CPU(ARM_HARDFP)
+#define ENABLE_LLINT 0
+#endif
+
 /* On some of the platforms where we have a JIT, we want to also have the 
    low-level interpreter. */
 #if !defined(ENABLE_LLINT) \
-- 
1.8.5.3

